make sure we re-use the existing CFLAGS settings so that we get all the right
flags (like -fno-stack-protector).  just append the few changes we want (the
optimization levels).

https://bugs.gentoo.org/590974

--- a/Makefile
+++ b/Makefile
@@ -52,10 +52,10 @@ reloc.o: reloc.c
 	$(CC) -c $(CFLAGS) -fno-strict-aliasing reloc.c
 
 test.o: test.c
-	$(CC) -c -Wall -march=i486 -m32 -O0 -fomit-frame-pointer -fno-builtin -ffreestanding test.c
+test.o: CFLAGS += -O0 -fno-PIC
 
 random.o: random.c
-	$(CC) -c -Wall -march=i486 -m32 -O3 -fomit-frame-pointer -fno-builtin -ffreestanding random.c
+random.o: CFLAGS += -O3 -fno-PIC
 	
 # rule for build number generation  
 build_number:
